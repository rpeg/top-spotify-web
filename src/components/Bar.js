import React from 'react';
import PropTypes from 'prop-types';
import { ProgressBar } from 'react-bootstrap';
import { makeSortedFrequencyArr } from '../lib/frequency';

const Bar = ({ items }) => {
  const sortedItems = makeSortedFrequencyArr(items);

  return (
    <ProgressBar style={{ height: '40px', backgroundColor: '#191414' }}>
      {sortedItems.map((item, i) => (
        <ProgressBar
          key={item.name}
          style={{ color: 'white', backgroundColor: `rgba(29, 185, 84, ${1 - (i * (1 / sortedItems.length))})` }}
          now={100 * (item.freq / items.length)}
          label={item.name}
        />
      ))}
    </ProgressBar>
  );
};

Bar.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Bar;
