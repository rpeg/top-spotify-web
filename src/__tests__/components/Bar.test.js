import React from 'react';
import { shallow } from 'enzyme';

import Bar from '../../components/Bar';

describe('Bar', () => {
  const items = ['a', 'b'];

  it('should not log errors in console', () => {
    const wrapper = shallow(<Bar items={items} />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Bar items={items} />);
    expect(wrapper).toMatchSnapshot();
  });
});
