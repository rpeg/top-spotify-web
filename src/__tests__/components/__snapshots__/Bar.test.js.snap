// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders correctly 1`] = `
<ProgressBar
  animated={false}
  isChild={false}
  max={100}
  min={0}
  srOnly={false}
  striped={false}
  style={
    Object {
      "backgroundColor": "#191414",
      "height": "40px",
    }
  }
>
  <ProgressBar
    animated={false}
    isChild={false}
    key="b"
    label="b"
    max={100}
    min={0}
    now={50}
    srOnly={false}
    striped={false}
    style={
      Object {
        "backgroundColor": "rgba(29, 185, 84, 1)",
        "color": "white",
      }
    }
  />
  <ProgressBar
    animated={false}
    isChild={false}
    key="a"
    label="a"
    max={100}
    min={0}
    now={50}
    srOnly={false}
    striped={false}
    style={
      Object {
        "backgroundColor": "rgba(29, 185, 84, 0.5)",
        "color": "white",
      }
    }
  />
</ProgressBar>
`;
